packages/composites/circuit-breaker/src/adapter.ts

```diff
@@ -0,0 +1,65 @@
+import { AdapterError, Requester, util, Validator, Logger } from '@chainlink/ea-bootstrap'
+import { AdapterRequest, ExecuteWithConfig, Config, ExecuteFactory } from '@chainlink/types'
+import { makeConfig } from './config'
+import { AxiosResponse } from 'axios'
+
+const customParams = {
+  primarySource: true,
+  secondarySource: true,
+}
+
+export interface ResponseSchema {
+  result: any
+}
+
+export const execute: ExecuteWithConfig<Config> = async (input, _, config) => {
+  const validator = new Validator(input, customParams)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.jobRunID
+  const primarySource = validator.validated.data.primarySource
+  const secondarySource = validator.validated.data.secondarySource
+  const sources = secondarySource ? [primarySource, secondarySource] : [primarySource]
+  const urls = sources.map((source) => util.getRequiredURL(source.toUpperCase()))
+  const response = await getResults(jobRunID, sources, urls, input, config)
+  response.data.result = Requester.validateResultNumber(response.data, ['data', 'result'])
+  return Requester.success(jobRunID, response)
+}
+
+export const makeExecute: ExecuteFactory<Config> = (config) => {
+  return async (request, context) => execute(request, context, config || makeConfig())
+}
+
+const getResults = async (
+  jobRunID: string,
+  sources: string[],
+  urls: string[],
+  request: AdapterRequest,
+  config: Config,
+): Promise<AxiosResponse<ResponseSchema>> => {
+  try {
+    Logger.info(`Trying to get result from ${sources[0]}`)
+    return await Requester.request<ResponseSchema>({
```

Comment:
I might have jumped the gun when requesting that we return this directly - have you verified what happens if the underlying source adapter returns 500 here?

Reply:
if the adapter fails, anyway a 500 error is returned, should it return any particular value? in this case would it be necessary to use a nested try catch?

Reply:
With the integration tests we have now I think this is good :+1: